src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -200,6 +206,14 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
     if (highestPriorityRoot) {
       nextPriorityLevel = highestPriorityLevel;
       priorityContext = nextPriorityLevel;
+
+      // Before we start any new work, let's make sure that we have a fresh
+      // stack to work from.
+      // TODO: This call is burried a bit too deep. It would be nice to have
+      // a single point which happens right before any new work and
+      // unfortunately this is it.
+      resetContextStack();
```

Comment:
Could we put the reassignment of `priorityContext` (line 208) inside `resetContextStack`? It's not quite the same as the other contexts but it serves a similar purpose.